geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/view/package.scala

```diff
@@ -45,4 +52,100 @@ package object view {
       case Some(f) => if (filter == Filter.INCLUDE) { f } else { andFilters(Seq(filter, f)) }
     }
   }
+
+  /**
+    * Read only data store - does not support creating/updating/deleting schemas or features
+    */
+  trait ReadOnlyDataStore extends DataStore {
+
+    private def error = throw new NotImplementedError("This data store is read-only")
+
+    override def createSchema(featureType: SimpleFeatureType): Unit = error
+    override def updateSchema(typeName: Name, featureType: SimpleFeatureType): Unit = error
+    override def updateSchema(typeName: String, featureType: SimpleFeatureType): Unit = error
+    override def removeSchema(typeName: Name): Unit = error
+    override def removeSchema(typeName: String): Unit = error
+    override def getFeatureWriter(typeName: String, transaction: Transaction): SimpleFeatureWriter = error
+    override def getFeatureWriter(typeName: String, filter: Filter, transaction: Transaction): SimpleFeatureWriter = error
+    override def getFeatureWriterAppend(typeName: String, transaction: Transaction): SimpleFeatureWriter = error
+  }
+
+  /**
+    * Data store that merges the schemas from multiple delegate stores and presents them as a unified result
+    *
+    * @param stores delegate stores
+    * @param namespace schema namespace
+    */
+  abstract class MergedDataStoreSchemas(stores: Seq[DataStore], namespace: Option[String])
```

Comment:
Are there other MDSS which we could be implementing?

Reply:
So far just the two :)